distributed_index_tests/lib/syncgateway.py

```diff
@@ -10,111 +10,16 @@
 
 class SyncGateway:
 
-    def __init__(self, host_info, db_name):
-        self.public_port = 4984
-        self.admin_port = 4985
-        self.ip = "http://{}".format(host_info["ip"])
-        self.host_name = host_info["name"]
-        self.db = Database(self.ip, self.admin_port, self.public_port, db_name)
+    def __init__(self, target):
+        self.ip = target["ip"]
+        self.url = "http://{}:4984".format(target["ip"])
```

Comment:
@ashvindersingh Perhaps the framework can consume ports from the ini file and we can abstract out the hard coded values? If you agree, with this approach, I would file an issue for it since the change may be decently sized refactor.